### DISCUSSION TITLE
Make Live Preview work with Commerce Variants

### DISCUSSION THREAD

**Discussion by angrybrad** [craftcms/commerce#2297] [Ideas]
> Created by: Gary Pearman ([gpearman@gmail.com](gpearman@gmail.com)) on 2016/07/18 11:49:43 +0000
Votes at time of UserVoice import: 3

------
I know this is a tricky one, as I've tried to hack the core to make this happen - definitely not a one liner.

The issue I'm having is that Live Preview doesn't show any variant fields when used in Commerce.

Also, the way I'm using variants allows different option entries on different variants, which are loaded by AJAX in the front end when a variant is selected. Meaning even if variant fields were shown, changing them in live preview wouldn't have any effect, as the AJAX call would load the old (previously saved values).

I'd love for this to be implemented as part of the core of Commerce, but I think it's going to be a PITA (hence why I think it's been left out in the first place).

**Comment by boboldehampsink** [comment]
At the moment it's even failing live preview for the whole product for me with the following message:

```
TypeError
Argument 1 passed to craft\commerce\elements\Product::setVariants() must be of the type array, null given, called in /app/user/vendor/craftcms/commerce/src/controllers/ProductsController.php on line 544
```

**Comment by lukeholder** [comment]
@boboldehampsink This has been fixed locally and I will be pushing to the develop branch shortly. 

Leaving ticket open for original request.

**Comment by j-greig** [comment]
Am I correct in thinking that `product.defaultVariant.stock` and `product.defaultVariant.hasStock` are not accessible in Live Preview?
